### ISSUE TITLE
Bug and extra behaviour in in plotExplanatoryVariables

### ISSUE THREAD

**Issue by LTLA** [davismcc/archive-scater#52]
This causes an error, presumably because some dimensions were dropped somewhere:

``` r
example(plotExplanatoryVariables)
plotExplanatoryVariables(example_sceset, variables=vars[1])
```

Also consider adding "density" as the y-axis label of this plot.

**Comment by davismcc** [comment]
Thanks for this. I've fixed the bug in commit 1d701bd . It was indeed caused by dropped dimensions, so a couple of additional "drop = FALSE" did the trick.

Also added "Density" label to y-axis of plot.

**Comment by davismcc** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;nMB6yJZf&#x2F;81-issue"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Issue</a>
